### ISSUE TITLE
NullReferenceException at Microsoft.AspNetCore.Http.DefaultHttpContext.get_Items()

### ISSUE THREAD

**Issue by gmajian** [dotnet/aspnetcore#29708]
#2799
We are seeing this issue https://github.com/dotnet/aspnetcore/issues/2799 randomly as well, but the previouls issue was closed automatically and lockked. So I have to create this new issue to get attention.

we have pretty much same call stack of the issue with https://github.com/dotnet/aspnetcore/issues/2799
We are using DotNetCore 2.1.1.0. Running on Windows Server 2016 with Docker.

`System.NullReferenceException:   at Microsoft.AspNetCore.Http.DefaultHttpContext.get_Items (Microsoft.AspNetCore.Http, Version=2.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60) 
`
Please let us know anything else we could provide to help the investigation.

**Comment by Tratcher** [comment]
This class of issue is almost always caused by unsafe multi-threaded use of the HttpContext in app code.

You can start by sharing the full stack trace. In theory you should be getting this error on two different stack traces and that should help you track down which components are running in parallel.
